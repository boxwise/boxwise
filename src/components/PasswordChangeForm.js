import React from "react";
import { Field, Formik } from "formik";
import { withStyles } from "@material-ui/core/styles";
import { TextField } from "formik-material-ui";
import Typography from "@material-ui/core/Typography";

import { handleError } from "utils";

import ButtonWithProgress from "./ButtonWithProgress";

const styles = theme => ({
  submit: {
    marginTop: theme.spacing.unit * 2
  }
});

const PasswordChangeForm = ({ classes, user, userPasswordChange }) => (
  <Formik
    initialValues={{
      currentPassword: "",
      newPassword: "",
      confirmedPassword: ""
    }}
    validate={values => {
      let errors = {};
      if (!values.currentPassword) {
        errors.currentPassword = "Enter your current password.";
      }
      if (!values.newPassword) {
        errors.newPassword = "Enter your new password.";
      }
      if (!values.confirmedPassword) {
        errors.confirmedPassword = "Confirmed your new password.";
      }
      if (values.newPassword !== values.confirmedPassword) {
        errors.mismatchedPassword =
          "The confirmed password is different from the new password.";
      } else if (values.newPassword.length < 6) {
        errors.malformedPassword = "Password should be at least 6 characters.";
      }

      return errors;
    }}
    onSubmit={(
      { currentPassword, newPassword },
      { setSubmitting, setErrors, setStatus }
    ) => {
      const email = user.data.email;
      userPasswordChange({ email, currentPassword, newPassword })
        .then(() => {
          setSubmitting(false);
          setStatus(0);
        })
        .catch(error => {
          setSubmitting(false);
          handleError(error);
          setErrors({ form: error.message });
        });
    }}
    render={({ handleSubmit, isSubmitting, errors, touched, status }) => (
      <form onSubmit={handleSubmit}>
        {status === 0 ? (
          <Typography color="primary">
            Your password has been updated successfully.
          </Typography>
        ) : null}

        {/* TODO: style errors */}
        {errors.form ? (
          <Typography color="error">{errors.form}</Typography>
        ) : null}

        {touched.currentPassword && errors.currentPassword ? (
          <Typography color="error">{errors.currentPassword}</Typography>
        ) : null}
        <Field
          type="password"
          label="Current Password"
          name="currentPassword"
          component={TextField}
          fullWidth
          margin="normal"
        />
        {touched.newPassword && errors.newPassword ? (
          <Typography color="error">{errors.newPassword}</Typography>
        ) : null}
        <Field
          type="password"
          label="New Password"
          name="newPassword"
          component={TextField}
          fullWidth
          margin="normal"
        />
        {touched.confirmedPassword && errors.confirmedPassword ? (
          <Typography color="error">{errors.confirmedPassword}</Typography>
        ) : null}
        <Field
          type="password"
          label="Confirm Password"
          name="confirmedPassword"
          component={TextField}
          fullWidth
          margin="normal"
        />
        {touched.newPassword &&
        touched.confirmedPassword &&
        errors.mismatchedPassword ? (
          <Typography color="error">{errors.mismatchedPassword}</Typography>
        ) : null}
        {touched.newPassword &&
        touched.confirmedPassword &&
        errors.malformedPassword ? (
          <Typography color="error">{errors.malformedPassword}</Typography>
        ) : null}
        <ButtonWithProgress
          variant="contained"
          color="primary"
          type="submit"
          loading={isSubmitting || user.isUpdating}
          className={classes.submit}
        >
          Update Password
        </ButtonWithProgress>
      </form>
    )}
  />
);

export default withStyles(styles)(PasswordChangeForm);
