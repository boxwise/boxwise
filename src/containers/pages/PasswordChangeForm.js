import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import PasswordChangePage from "pages/PasswordChangePage";

const mapStateToProps = ({ user }) => ({ user });
export default connect(mapStateToProps)(withRouter(PasswordChangePage));
