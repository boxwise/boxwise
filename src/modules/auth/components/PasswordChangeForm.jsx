import React, { useState } from "react";
import Typography from "@material-ui/core/Typography";

import { useMaterialUIForm } from "hooks/forms";
import SubmitButton from "components/SubmitButton";
import FormErrorText from "components/FormErrorText";
import PasswordField from "components/PasswordField";

export const PasswordChangeForm = ({ serverError, userPasswordChange }) => {
  const handleValidation = values => {
    const errors = {};
    if (!values.currentPassword) {
      errors.currentPassword = "Enter your current password.";
    }
    if (!values.newPassword) {
      errors.newPassword = "Enter your new password.";
    }
    if (!values.confirmedPassword) {
      errors.confirmedPassword = "Please confirm your new password.";
    } else if (values.newPassword !== values.confirmedPassword) {
      errors.confirmedPassword =
        "The confirmed password is different from the new password.";
    } else if (values.newPassword.length < 6) {
      errors.newPassword = "Password should be at least 6 characters.";
    }
    return errors;
  };
  const hasSubmittedForm = false;
  // const [hasSubmittedForm, setHasSubmittedForm] = useState(false);
  const { attachValidation, handleSubmit, isSubmitting } = useMaterialUIForm(
    userPasswordChange,
    handleValidation
  );
  // .then(() => setHasSubmittedForm(true));
  return (
    <form onSubmit={handleSubmit}>
      {hasSubmittedForm && !serverError ? (
        <Typography color="primary" data-testid="pwdChangeConfirmation">
          Your password has been updated successfully.
        </Typography>
      ) : null}
      <FormErrorText message={serverError} />
      <PasswordField
        name="currentPassword"
        label="Current Password"
        dataTestId="currentPassword"
        {...attachValidation("currentPassword")}
      />
      <PasswordField
        name="newPassword"
        label="New Password"
        dataTestId="newPassword"
        {...attachValidation("newPassword")}
      />
      <PasswordField
        name="confirmedPassword"
        label="Confirm Password"
        dataTestId="confirmedPassword"
        {...attachValidation("confirmedPassword")}
      />
      <SubmitButton
        isSubmitting={isSubmitting}
        data-testid="updatePasswordButton"
      >
        Update Password
      </SubmitButton>
    </form>
  );
};

export default PasswordChangeForm;
