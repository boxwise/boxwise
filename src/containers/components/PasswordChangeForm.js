import { connect } from "react-redux";

import { userPasswordChange } from "../../actions/auth";
import PasswordChangeForm from "../../components/PasswordChangeForm";

const mapStateToProps = ({ user }) => ({ user });
const mapDispatchToProps = { userPasswordChange };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PasswordChangeForm);
