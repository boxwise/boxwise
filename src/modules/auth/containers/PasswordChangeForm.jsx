import { connect } from "react-redux";

import PasswordChangeForm from "../components/PasswordChangeForm";
import { userPasswordChange } from "../actions";

const mapStateToProps = ({ user }) => ({
  serverError: user.error && user.error.message
});
const mapDispatchToProps = { userPasswordChange };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PasswordChangeForm);
