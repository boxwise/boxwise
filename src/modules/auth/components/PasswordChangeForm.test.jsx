import React from "react";
import { render, fireEvent } from "react-testing-library";

import { PasswordChangeForm } from "./PasswordChangeForm";

const typeText = (element, value) => {
  fireEvent.change(element, { target: { value } });
};

describe("PasswordChangeForm", () => {
  let container;
  let getByText;
  let getByTestId;

  const handler = jest.fn(() => Promise.resolve());
  const clickSubmit = () => fireEvent.click(getByText(/Update Password/i));
  const enterCurrentPassword = text =>
    typeText(getByTestId("currentPassword"), text);
  const enterNewPassword = text => typeText(getByTestId("newPassword"), text);
  const enterConfirmedPassword = text =>
    typeText(getByTestId("confirmedPassword"), text);

  beforeEach(() => {
    ({ container, getByText, getByTestId } = render(
      <PasswordChangeForm userPasswordChange={handler} />
    ));
  });

  it("does not submit when password is not provided", () => {
    enterCurrentPassword("xx");
    clickSubmit();

    expect(container).toMatchSnapshot();
    expect(handler).not.toBeCalled();
  });

  it("submits when password is provided at least 6 characters", () => {
    typeText(getByTestId("currentPassword"), "hellooooooooo");
    // enterCurrentPassword("abc");
    enterNewPassword("xyzxyz9");
    enterConfirmedPassword("xyzxyz9");

    expect(container).toMatchSnapshot("x");

    // clickSubmit();

    expect(container).toMatchSnapshot("y");
    expect(handler).toBeCalledWith({
      currentPassword: "abc",
      confirmedPassword: "xyzxyz9",
      newPassword: "xyzxyz9"
    });
  });
  // password updated successfully

  // it("renders correctly", () => {
  //   expect(container).toMatchSnapshot();
  // });
});
