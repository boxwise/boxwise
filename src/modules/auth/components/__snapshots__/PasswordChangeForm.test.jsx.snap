// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordChangeForm does not submit when password is not provided 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    >
      <label
        class="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12"
        data-shrink="true"
        for="currentPassword"
      >
        Current Password
      </label>
      <div
        class="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
          data-testid="currentPassword"
          id="currentPassword"
          name="currentPassword"
          type="password"
          value="xx"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    >
      <label
        class="MuiFormLabel-root-16 MuiFormLabel-error-19 MuiInputLabel-error-8 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
        data-shrink="false"
        for="newPassword"
      >
        New Password
      </label>
      <div
        class="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-error-42 MuiInput-error-28 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
      >
        <input
          aria-describedby="newPassword-helper-text"
          aria-invalid="true"
          class="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
          data-testid="newPassword"
          id="newPassword"
          name="newPassword"
          type="password"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root-93 MuiFormHelperText-error-94"
        id="newPassword-helper-text"
      >
        Enter your new password.
      </p>
    </div>
    <div
      class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    >
      <label
        class="MuiFormLabel-root-16 MuiFormLabel-error-19 MuiInputLabel-error-8 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
        data-shrink="false"
        for="confirmedPassword"
      >
        Confirm Password
      </label>
      <div
        class="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-error-42 MuiInput-error-28 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
      >
        <input
          aria-describedby="confirmedPassword-helper-text"
          aria-invalid="true"
          class="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
          data-testid="confirmedPassword"
          id="confirmedPassword"
          name="confirmedPassword"
          type="password"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root-93 MuiFormHelperText-error-94"
        id="confirmedPassword-helper-text"
      >
        Please confirm your new password.
      </p>
    </div>
    <div
      class="SubmitButton-root-53 SubmitButton-submitInPage-56"
    >
      <button
        class="MuiButtonBase-root-83 MuiButton-root-57 MuiButton-contained-68 MuiButton-containedPrimary-69 MuiButton-raised-71 MuiButton-raisedPrimary-72"
        data-testid="updatePasswordButton"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-58"
        >
          Update Password
        </span>
        <span
          class="MuiTouchRipple-root-86"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`PasswordChangeForm renders correctly 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root-246 MuiFormControl-marginNormal-247 MuiFormControl-fullWidth-249"
    >
      <label
        class="MuiFormLabel-root-261 MuiInputLabel-root-250 MuiInputLabel-formControl-255 MuiInputLabel-animated-258"
        data-shrink="false"
        for="currentPassword"
      >
        Current Password
      </label>
      <div
        class="MuiInputBase-root-281 MuiInput-root-268 MuiInput-underline-272 MuiInputBase-fullWidth-290 MuiInput-fullWidth-275 MuiInputBase-formControl-282 MuiInput-formControl-269"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-291 MuiInput-input-276 MuiInputBase-inputType-294 MuiInput-inputType-279"
          data-testid="currentPassword"
          id="currentPassword"
          name="currentPassword"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-246 MuiFormControl-marginNormal-247 MuiFormControl-fullWidth-249"
    >
      <label
        class="MuiFormLabel-root-261 MuiInputLabel-root-250 MuiInputLabel-formControl-255 MuiInputLabel-animated-258"
        data-shrink="false"
        for="newPassword"
      >
        New Password
      </label>
      <div
        class="MuiInputBase-root-281 MuiInput-root-268 MuiInput-underline-272 MuiInputBase-fullWidth-290 MuiInput-fullWidth-275 MuiInputBase-formControl-282 MuiInput-formControl-269"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-291 MuiInput-input-276 MuiInputBase-inputType-294 MuiInput-inputType-279"
          data-testid="newPassword"
          id="newPassword"
          name="newPassword"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-246 MuiFormControl-marginNormal-247 MuiFormControl-fullWidth-249"
    >
      <label
        class="MuiFormLabel-root-261 MuiInputLabel-root-250 MuiInputLabel-formControl-255 MuiInputLabel-animated-258"
        data-shrink="false"
        for="confirmedPassword"
      >
        Confirm Password
      </label>
      <div
        class="MuiInputBase-root-281 MuiInput-root-268 MuiInput-underline-272 MuiInputBase-fullWidth-290 MuiInput-fullWidth-275 MuiInputBase-formControl-282 MuiInput-formControl-269"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-291 MuiInput-input-276 MuiInputBase-inputType-294 MuiInput-inputType-279"
          data-testid="confirmedPassword"
          id="confirmedPassword"
          name="confirmedPassword"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="SubmitButton-root-298 SubmitButton-submitInPage-301"
    >
      <button
        class="MuiButtonBase-root-328 MuiButton-root-302 MuiButton-contained-313 MuiButton-containedPrimary-314 MuiButton-raised-316 MuiButton-raisedPrimary-317"
        data-testid="updatePasswordButton"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-303"
        >
          Update Password
        </span>
        <span
          class="MuiTouchRipple-root-331"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`PasswordChangeForm submits when password is provided at least 6 characters: after submit 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112"
        data-shrink="true"
        for="currentPassword"
      >
        Current Password
      </label>
      <div
        class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 MuiInputBase-formControl-137 MuiInput-formControl-124"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
          data-testid="currentPassword"
          id="currentPassword"
          name="currentPassword"
          type="password"
          value="abc"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112"
        data-shrink="true"
        for="newPassword"
      >
        New Password
      </label>
      <div
        class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 MuiInputBase-formControl-137 MuiInput-formControl-124"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
          data-testid="newPassword"
          id="newPassword"
          name="newPassword"
          type="password"
          value="xyzxyz9"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112"
        data-shrink="true"
        for="confirmedPassword"
      >
        Confirm Password
      </label>
      <div
        class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 MuiInputBase-formControl-137 MuiInput-formControl-124"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
          data-testid="confirmedPassword"
          id="confirmedPassword"
          name="confirmedPassword"
          type="password"
          value="xyzxyz9"
        />
      </div>
    </div>
    <div
      class="SubmitButton-root-153 SubmitButton-submitInPage-156"
    >
      <button
        class="MuiButtonBase-root-183 MuiButtonBase-disabled-184 MuiButton-root-157 MuiButton-contained-168 MuiButton-containedPrimary-169 MuiButton-raised-171 MuiButton-raisedPrimary-172 MuiButton-disabled-177"
        data-testid="updatePasswordButton"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label-158"
        >
          Update Password
        </span>
      </button>
      <div
        class="MuiCircularProgress-root-193 MuiCircularProgress-colorPrimary-196 MuiCircularProgress-indeterminate-195 SubmitButton-progress-154"
        role="progressbar"
        style="width: 24px; height: 24px;"
      >
        <svg
          class="MuiCircularProgress-svg-198"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle-199 MuiCircularProgress-circleIndeterminate-201"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </form>
</div>
`;

exports[`PasswordChangeForm submits when password is provided at least 6 characters: before submit 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112"
        data-shrink="true"
        for="currentPassword"
      >
        Current Password
      </label>
      <div
        class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 MuiInputBase-formControl-137 MuiInput-formControl-124"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
          data-testid="currentPassword"
          id="currentPassword"
          name="currentPassword"
          type="password"
          value="abc"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112"
        data-shrink="true"
        for="newPassword"
      >
        New Password
      </label>
      <div
        class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 MuiInputBase-formControl-137 MuiInput-formControl-124"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
          data-testid="newPassword"
          id="newPassword"
          name="newPassword"
          type="password"
          value="xyzxyz9"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root-101 MuiFormControl-marginNormal-102 MuiFormControl-fullWidth-104"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112"
        data-shrink="true"
        for="confirmedPassword"
      >
        Confirm Password
      </label>
      <div
        class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 MuiInputBase-formControl-137 MuiInput-formControl-124"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
          data-testid="confirmedPassword"
          id="confirmedPassword"
          name="confirmedPassword"
          type="password"
          value="xyzxyz9"
        />
      </div>
    </div>
    <div
      class="SubmitButton-root-153 SubmitButton-submitInPage-156"
    >
      <button
        class="MuiButtonBase-root-183 MuiButton-root-157 MuiButton-contained-168 MuiButton-containedPrimary-169 MuiButton-raised-171 MuiButton-raisedPrimary-172"
        data-testid="updatePasswordButton"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-158"
        >
          Update Password
        </span>
        <span
          class="MuiTouchRipple-root-186"
        />
      </button>
    </div>
  </form>
</div>
`;
